src/util/strutil.h

```diff
@@ -7,19 +7,41 @@
 #pragma once
 
 // return utf8 encoded length of string
-int str_length(const char *str);
+int str_length
+(
+	const char *str
+);
 
-//convert str to a lower case string and save it in lower
-void str_tolower(const char *str, char *lower, size_t *lower_len);
+// return true if utf8 string is valid
+bool str_utf8_validate
+(
+    const char *str
+);
 
-//convert str to a n upper case string and save it in upper
-void str_toupper(const char *str, char *upper, size_t *upper_len);
+// convert ascii str to a lower case string and save it in lower
+void str_tolower_ascii
+(
+	const char *str,
+	char *lower,
+	size_t *lower_len
+);
 
+// return utf8 string converted to lower case
+char *str_tolower
+(
+	const char *str
+);
 
-// Utility function to increase the size of a buffer.
-void str_ExtendBuffer(
+// return utf8 string converted to upper case
+char *str_toupper
+(
+	const char *str
+);
+
+// utility function to increase the size of a buffer
+void str_ExtendBuffer
+(
 	char **buf,           // buffer to populate
 	size_t *bufferLen,    // size of buffer
 	size_t extensionLen   // number of bytes to add
 );
```

Comment:
Why do we need this function, who uses it?
isn't this a simple call to `realloc` ?